include/soll/AST/Expr.h

```diff
@@ -107,16 +107,16 @@ class Identifier : public Expr {
   }
   const Token &getToken() const { return T; }
   llvm::StringRef getName() const { return T.getIdentifierInfo()->getName(); }
-  void setCorrespondDecl(Decl *D) {
-    this->D = D;
-    updateTypeFromCurrentDecl();
-  }
-  Decl *getCorrespondDecl() { return std::get<Decl *>(D); }
-  const Decl *getCorrespondDecl() const { return std::get<Decl *>(D); }
+  void setCorrespondDecl(Decl *D) { this->D = D; }
+  Decl *getCorrespondDecl() { return *std::get_if<Decl *>(&D); }
+  const Decl *getCorrespondDecl() const { return *std::get_if<Decl *>(&D); }
```

Comment:
The `std::get_if` should return `Decl *` or `nullptr`. Do we need to dereference it?